packages/babel-traverse/src/path/introspection.js

```diff
@@ -439,3 +439,29 @@ export function isConstantExpression() {
 
   return false;
 }
+
+export function isInStrictMode() {
+  const start = this.isProgram() ? this : this.parentPath;
+
+  const strictParent = start.find(path => {
+    if (path.isProgram({ sourceType: "module" })) return true;
+
+    if (path.isClass()) return true;
+
+    if (!path.isProgram() && !path.isFunction()) return false;
+
+    let { node } = path;
+    if (path.isFunction()) node = node.body;
+
+    const { directives } = node;
+    if (!directives) return false;
```

Comment:
I think in theory this is supposed to always exist unless `node` isn't `BlockStatement` (because it is an arrow body expression)? Might be clearer to do

```
if (!path.isProgram() && !path.isFunction()) return false;

if (path.isArrowFunction() && !path.get("body").isBlockStatement()) return false;
```
then remove this check?

Reply:
Yup, that was it. Done.